linked_list.h

```diff
@@ -102,17 +118,16 @@ int insert_node(void *data, char type, size_t bytes, list **head) {
 }
 
 int insert_char(char data, list ** head) {
-	return insert_node((char *)&data, 'c', sizeof(char), head);
+	return insert_node((char *)&data, 'c', &get_size, head);

```

Comment:
Dont take type info again. Remove those 'c', etc. you already know type here as you called insert_char
Here this should be changed from 
insert_node((char *)&data, 'c', &get_size, head);
to 
int getCharSize(void *data) {
    return 1;
}

int getStringSize(void *data) {
    return strlen((char *)data));
}
//For char
insert_node((void *)&data, &getCharSize,&printChar, head);
//For string inside insert_string method
insert_node((void *)&data, &getStringSize, &printStr, head);

Remove that switch case in insert as well, It should be independent. Like in future if you add a new struct type to linked list, insert_node method should not change, only wrappers will change.

Reply:
removed the switch, added new functions, changed the typedef for get_size